src/segmantic/seg/evaluation.py

```diff
@@ -0,0 +1,242 @@
+import numpy as np
+import itk
+import random
+import colorsys
+from matplotlib import colors
+from pathlib import Path
+from typing import Dict, List, Optional
+
+from segmantic.prepro.core import itkImage
+
+
+def hausdorff_surface_distance(y_pred: itkImage, y_ref: itkImage) -> Dict[str, float]:
+    """Compute symmetric surface distances between two binary masks
+
+    Args:
+        y_pred (itkImage): predicted segmentation
+        y_ref (itkImage): reference segmentation
+
+    Returns:
+        Dict[str, float]: keys are 'mean', 'median', 'std', 'max'
+    """
+    seg_surface = itk.binary_contour_image_filter(y_pred, foreground_value=1)
+    ref_surface = itk.binary_contour_image_filter(y_ref, foreground_value=1)
+
+    seg_distance = itk.signed_maurer_distance_map_image_filter(
+        y_pred, use_image_spacing=True, squared_distance=False, inside_is_positive=False
+    )
+    ref_distance = itk.signed_maurer_distance_map_image_filter(
+        y_ref, use_image_spacing=True, squared_distance=False, inside_is_positive=False
+    )
+
+    # get distance at contour of foreground label
+    ref2seg_distance = np.multiply(ref_surface, seg_distance)
+    ref2seg_distance = ref2seg_distance[np.nonzero(ref_surface)]
+
+    seg2ref_distance = np.multiply(seg_surface, ref_distance)
+    seg2ref_distance = seg2ref_distance[np.nonzero(seg_surface)]
+
+    # compute statistics on symmetric distances (both directions)
+    all_surface_distances = np.concatenate(
+        (ref2seg_distance, seg2ref_distance), axis=None
+    )
+    all_surface_distances = np.abs(all_surface_distances)
+
+    mean_surface_distance = np.mean(all_surface_distances)
+    median_surface_distance = np.median(all_surface_distances)
+    std_surface_distance = np.std(all_surface_distances)
+    max_surface_distance = np.max(all_surface_distances)
+    return {
+        "mean": mean_surface_distance,
+        "median": median_surface_distance,
+        "std": std_surface_distance,
+        "max": max_surface_distance,
+    }
+
+
+def hausdorff_pointwise_distance(y_pred: itkImage, y_ref: itkImage) -> Dict[str, float]:
+    """Compute symmetric point-wise distances between two binary masks
+
+    Args:
```

Comment:
I wonder if the types in this doc are really necessary provided that they are in the annotation. I guess you did it with an automatic doc generator? It is kind of redundant

Reply:
positional arguments (without default assigned) don't get any type documentation, unless you specify e.g. 
```
y: SomeType = typer.Option(..., "-y")
```
so
1. yes, I used an automatic documentation tool that adds types
2. I kept them because I did not know yet about the `typer.Option(..., ` way of documenting arguments